config.js

```diff
@@ -64,62 +63,74 @@ var getMessage = function(value, locale) {
   return result;
 };
 
+var loadSettings = function(callback) {
+  db.medic.get('_design/medic', function(err, ddoc) {
+    if (err) {
+      return callback(err);
+    }
+    settings = ddoc.app_settings;
+    _.defaults(settings, defaults);
+    callback();
+  });
+};
+
+var loadTranslations = function() {
+  var options = { key: [ 'translations' ], include_docs: true };
+  db.medic.view('medic', 'doc_by_type', options, function(err, result) {
+    if (err) {
+      console.error('Error loading translations - starting up anyway', err);
+      return;
+    }
+    result.rows.forEach(function(row) {
+      translations[row.doc.code] = row.doc.values;
+    });
+  });
+};
+
 module.exports = {
   get: function(key) {
     return settings[key];
   },
   translate: function(key, locale, ctx) {
-    var value;
-    ctx = ctx || {};
-
     if (_.isObject(locale)) {
       ctx = locale;
       locale = null;
     }
-
+    locale = locale || (settings && settings.locale) || 'en';
     if (_.isObject(key)) {
       return getMessage(key, locale) || key;
     }
-
-    value = _.findWhere(settings.translations, { key: key });
-
-    value = getMessage(value, locale) || key;
-
+    var value = (translations[locale] && translations[locale][key]) ||
+                (translations.en && translations.en[key]) ||
+                key;
     // underscore templates will return ReferenceError if all variables in
     // template are not defined.
     try {
-      return _.template(value)(ctx);
+      return _.template(value)(ctx || {});
     } catch(e) {
       return value;
     }
   },
   load: function(callback) {
-    db.medic.get('_design/medic', function(err, ddoc) {
-      if (err) {
-        return callback(err);
-      }
-      settings = ddoc.app_settings;
-      _.defaults(settings, defaults);
-      callback();
-    });
+    loadSettings(callback);
+    loadTranslations();
```

Comment:
I'm happy with translations loading in the background. We can't callback until the settings are loaded because of permissions and so on, but I'm happy for translations to be loaded just as soon as possible without blocking.